### DISCUSSION TITLE
Trying to understand memory - help appreciated

### DISCUSSION THREAD

**Discussion by n9jcv** [picocomputer/community#75] [Q&A] [answered]
Here is what I did.  Fresh boot.  Then I entered data at memory loc 0001, 0002 and 0003.  Display memory 
![20240111_202623](https://github.com/picocomputer/community/assets/6472203/8e649f81-e2fa-4642-979f-e33b9a38f0da)

Looks good.
Now I press the reboot button by the vga pico.
![20240111_202639](https://github.com/picocomputer/community/assets/6472203/bdfeaa14-d322-4cdf-9a17-509e2ddc276c)

At this point I expected to lose contents of 0001, 0002 and 0003.  Not sure why it still retains my values.

Then I actually pull the power and power up again.  
![20240111_202758](https://github.com/picocomputer/community/assets/6472203/791ad100-00b9-4cf4-be08-a8d0492fbdd6)

Now I have lost the data at 0001, 0002 and 0003.

Why did a reboot not clear the memory like a power reset?

Thanks all!

**Comment by WojciechGw** [comment, marked as answer]
rp6502 memory chip in 65C02 ecosystem is SRAM
pressing the reset button does not cut power to the memory chip, so its contents remain preserved after this action
rp6502 initialization routines do not affect the contents of SRAM memory chip (except for its last page)

**Reply by n9jcv** [reply]
Thank you!
